common.py

```diff
@@ -110,30 +110,53 @@ class TestMetadata:
     server: PodInfo
     client: PodInfo
 
-    def __post_init__(self) -> None:
-        self.test_case_id = enum_convert(TestCaseType, self.test_case_id)
-        self.test_type = enum_convert(TestType, self.test_type)
-        if isinstance(self.server, dict):
-            self.server = dataclass_from_dict(PodInfo, self.server)
-        if isinstance(self.client, dict):
-            self.client = dataclass_from_dict(PodInfo, self.client)
+    def __init__(
```

Comment:
why was this changed from post_init to init?

Reply:
in a simplified form, the code previously was
```python
@dataclass
class TestMetadata:
    test_type: TestType
    def __post_init__(self):
        self.test_type = enum_convert(TestType, self.test_type)
```

In this case, `TestMetadata` has signature `test_type: TestType`. It makes no sense to afterwards in `__post_init__()` expect that it also could be a `str | int` and do an `enum_convert()`.

That inconsistency can be resolved in two ways:

```python
class TestMetadata:
    test_type: TestType
```
here you don't need an `__post_init__()`. All callers must pass a `TestType` argument. If they have a string at hand, the callers must convert it.

Instead I did:
```python
class TestMetadata:
    test_type: TestType
    def __init__(self, test_type: TestType | str | int):
        self.test_type = enum_convert(TestType, test_type)
```
Now you can construct the instance with a `test_type: TestType | str | int`, but `instance.test_type` will then be normalized/converted to an `TestType` enum.

Both works. The former pushes the work of converting the type to the (potentially many) callers. It also pushes the knowledge of how to get from a `str|int` to a `TestType` to those callers. I think it's better if that complexity is at one place.

Granted, few callers actually use this feature/flexiblity. Version 1 would work too.

Reply:
Got it, thanks!